modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/sampledata/SampleDataExecutor.java

```diff
@@ -81,9 +83,9 @@ public SampleDataResult getSampleData(ComponentModel componentModel, ComponentEl
       if (cause instanceof SampleDataException) {
         failureBuilder.withFailureCode(((SampleDataException) cause).getFailureCode());
       }
```

Comment:
This failure is not needed anymore since the exception will be thrown

Reply:
No, actually the way the connector has to return a "failure" is by throwing this exception which has a an error code. If any other exception is thrown we should propagate it but this transformation and exception handling is needed here.